sonic_platform_base/chassis_base.py

```diff
@@ -463,6 +488,62 @@ def get_status_led(self):
         """
         raise NotImplementedError
 
+    ##############################################
+    # Card methods
+    ##############################################
+
+    def get_num_cards(self):
+        """
+        Retrieves the number of cards available on this chassis
+
+        Returns:
+            An integer, the number of cards available on this chassis
+        """
+        return len(self._card_list)
+
+    def get_all_cards(self):
+        """
+        Retrieves all cards available on this chassis
+
+        Returns:
+            A list of objects derived from CardBase representing all
+            cards available on this chassis
+        """
+        return self._card_list
+
+    def get_card(self, index):
+        """
+        Retrieves card represented by (0-based) index <index>
+
+        Args:
+            index: An integer, the index (0-based) of the card to retrieve
+
+        Returns:
+            An object dervied from CardBase representing the specified card
+        """
+        card = None
+
+        try:
+            card = self._card_list[index]
+        except IndexError:
+            sys.stderr.write("Card index {} out of range (0-{})\n".format(
+                             index, len(self._card_list)-1))
+
+        return card
+
+    def get_card_index(self, instance, module_type):
+        """
+        Retrieves card represented by <instance, module-type>
```

Comment:
what module_type do we have here, can you clarify in the description?

Reply:
This API will be fixed based on review comments. It will be get_module_index(self, module_name)